src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -542,8 +542,15 @@ public SymbolInfo GetSymbolInfo(ExpressionSyntax expression, CancellationToken c
                     case SyntaxKind.ParenthesizedVariableDesignation:
                         if (((TypeSyntax)expression).IsVar)
                         {
-                            return SymbolInfo.None;
+                            var varTypeInfo = GetTypeInfoWorker(expression, cancellationToken);
+                            if (varTypeInfo.Type.TypeKind != TypeKind.Error)
```

Comment:
>Type [](http://example.com/codeflow?start=44&length=4)

It looks like ```Type``` could be null

Reply:
@AlekseyTs I thought if the type cannot be determined we'll get back an error type, not null. Can you elaborate more on the scenario that can return null?

My assumption is that `TypeExpressionSyntax` will bind to `BoundTypeExpression`, which has non-null type.

https://github.com/dotnet/roslyn/blob/2d7ab4a8fd8f55366f7e064b7254d1823f086d6a/src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml#L268-L269

This should be what semantic model returns.

More specifically, in `CSharpSemanticModel.GetTypeInfoForNode`, we'll get into this code path:

```csharp
                if (boundExpr.HasExpressionType())
                {
                    type = boundExpr.Type;
                    // .....
                }
```

Reply:
> I thought if the type cannot be determined we'll get back an error type, not null. Can you elaborate more on the scenario that can return null?

In general, the ```Type``` in the structure can be null. It will be more robust to handle this potential situation regardless of what the current implementation of ```GetTypeInfoWorker``` does today for the specific scenario. Things can change in the future, it is legal for the property to be null, there is really no requirement to return an error type symbol instead.